src/code/z_play.c

```diff
@@ -74,39 +74,206 @@ s32 Play_InCsMode(PlayState* this) {
+        camera->atActorOffset.x = at->x - camera->focalActor->world.pos.x;
+        camera->atActorOffset.y = at->y - camera->focalActor->world.pos.y;
+        camera->atActorOffset.z = at->z - camera->focalActor->world.pos.z;
+    } else {
+        camera->atActorOffset.x = camera->atActorOffset.y = camera->atActorOffset.z = 0.0f;
+    }
+
+    camera->atLerpStepScale = 0.01f;
+
+    return successfullySet;
+}
+
+s32 Play_SetCameraAtEyeUp(PlayState* this, s16 camId, Vec3f* at, Vec3f* eye, Vec3f* up) {
+    s32 successfullySet = 0;
+    s16 camIdx = (camId == CAM_ID_NONE) ? this->activeCamId : camId;
+    Camera* camera = this->cameraPtrs[camIdx];
+
+    successfullySet |= Camera_SetViewParam(camera, CAM_VIEW_AT, at);
+    successfullySet <<= 1;
+    successfullySet |= Camera_SetViewParam(camera, CAM_VIEW_EYE, eye);
+    successfullySet <<= 1;
+    successfullySet |= Camera_SetViewParam(camera, CAM_VIEW_UP, up);
+
+    camera->dist = Math3D_Distance(at, eye);
+
+    if (camera->focalActor != NULL) {
+        camera->atActorOffset.x = at->x - camera->focalActor->world.pos.x;
+        camera->atActorOffset.y = at->y - camera->focalActor->world.pos.y;
+        camera->atActorOffset.z = at->z - camera->focalActor->world.pos.z;
+    } else {
+        camera->atActorOffset.x = camera->atActorOffset.y = camera->atActorOffset.z = 0.0f;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_play/Play_CameraGetUID.s")
+    camera->atLerpStepScale = 0.01f;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_play/func_80169BF8.s")
+    return successfullySet;
+}
+
+s32 Play_SetCameraFov(PlayState* this, s16 camId, f32 fov) {
+    s32 successfullySet = Camera_SetViewParam(this->cameraPtrs[camId], CAM_VIEW_FOV, &fov) & 1;
+
+    if (1) {}
+    return successfullySet;
+}
+
+s32 Play_SetCameraRoll(PlayState* this, s16 camId, s16 roll) {
+    s16 camIdx = (camId == CAM_ID_NONE) ? this->activeCamId : camId;
+    Camera* camera = this->cameraPtrs[camIdx];
+
+    camera->roll = roll;
+
+    return 1;
+}
+
+void Play_CopyCamera(PlayState* this, s16 destCamId, s16 srcCamId) {
+    s16 srcCamId2 = (srcCamId == CAM_ID_NONE) ? this->activeCamId : srcCamId;
+    s16 destCamId1 = (destCamId == CAM_ID_NONE) ? this->activeCamId : destCamId;
+
+    Camera_Copy(this->cameraPtrs[destCamId1], this->cameraPtrs[srcCamId2]);
+}
+
+// Same as Play_ChangeCameraSetting but also calls Camera_InitPlayerSettings
+s32 func_80169A50(PlayState* this, s16 camId, Player* player, s16 setting) {
+    Camera* camera;
+    s16 camIdx = (camId == CAM_ID_NONE) ? this->activeCamId : camId;
+
+    camera = this->cameraPtrs[camIdx];
+    Camera_InitPlayerSettings(camera, player);
+    return Camera_ChangeSetting(camera, setting);
+}
+
+s32 Play_ChangeCameraSetting(PlayState* this, s16 camId, s16 setting) {
+    return Camera_ChangeSetting(Play_GetCamera(this, camId), setting);
+}
+
+// Related to bosses and fishing
+void func_80169AFC(PlayState* this, s16 camId, s16 timer) {
```

Comment:
Can we say any more since we have at least one boss finished and documented now?